File: src/server/files/models/File.model.ts
import { Column, DataType, HasMany,Scopes, Table, } from 'sequelize-typescript';

import { AbstractModel, } from '../../database/AbstractModel';
import { FileUser, } from './FileUser.model';

@Scopes(() => ({
  defaultScope: {
    attributes: {
      exclude: ['createdAt', 'updatedAt'],
    },
  },
}))
@Table({})
export class File extends AbstractModel {
  @Column({
    type: DataType.STRING(10),
    allowNull: false,
  })
  declare ext: string;

  @Column({
    type: DataType.STRING,
    allowNull: false,
  })
  declare mime: string;

  @Column({
    type: DataType.BIGINT,
    allowNull: false,
  })
  declare size: number;

  @Column({
    type: DataType.STRING,
    allowNull: false,
  })
  declare storage: string;

  @Column({
    type: DataType.STRING(),
    allowNull: false,
  })
  declare hash: string;

  @HasMany(() => FileUser)
    fileUsers?: FileUser[];
}

File: app/files/models/file.model.go
package files

import (
	"file-service/app/database"
)

type File struct {
	database.AbstractModel
	EXT     string `gorm:"type:varchar(10);column:ext;not null"`
	MIME    string `gorm:"type:varchar(255);column:mime;not null"`
	Size    uint32 `gorm:"type:bigint;not null"`
	Storage string `gorm:"type:varchar(255);not null"`
	Hash    string `gorm:"type:varchar(255);not null"`
	FileUsers *[]FileUser
}
